### PR TITLE
I3S-Debug - Compare geometry bounding volume vs tile bounding volume

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
image

### CODE REVIEW

**Comment by dryabinin-actionengine** [comment]
@ibgreen Also there is a fix for scaling process in math.gl - [PR](https://github.com/uber-web/math.gl/pull/206/checks?check_run_id=2323707840)

**Review by ibgreen — APPROVED** [review]

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js [hunk 1]
```diff
@@ -169,6 +175,23 @@ export default class TileValidator extends PureComponent {
     this.setState({triangleMessages});
   }
 
+  compareGeometryBoundingVolumeVsTileBoundingVolume(tile) {
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/components/tile-validator.js:178]: Why are we checking separately for null and falsy? This seems like a bad API.

File: examples/experimental/i3s-17-and-debug/tile-debug.js [hunk 2]
```diff
@@ -334,3 +315,80 @@ function isAllVerticesInsideBoundingVolume(boundingVolume, positions) {
 
   return isVerticesInsideObb;
 }
+
+export function isTileGeometryInsideBoundingVolume(tile) {
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/tile-debug.js:319]: No JSDoc/types?

No test cases?

**Review by ibgreen — APPROVED** [review]

File: examples/experimental/i3s-17-and-debug/tile-debug.js (hunk 2 above)
Comment (dryabinin-actionengine) [reply examples/experimental/i3s-17-and-debug/tile-debug.js:319]: [Refactoring PR](https://github.com/visgl/loaders.gl/pull/1323)
It is not possible to implement tests outside of modules folder for now. We should investigate on that and also cover this code by tests.

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply examples/experimental/i3s-17-and-debug/components/tile-validator.js:178]: Refactoring here [Refactoring PR](https://github.com/visgl/loaders.gl/pull/1323)

**Pull request merged**